decidim-core/app/controllers/decidim/timeouts_controller.rb

```diff
@@ -5,11 +5,10 @@
 module Decidim
   # Tells/Extends time before inactivity warning or automatic logout.
   class TimeoutsController < Decidim::ApplicationController
-    prepend_before_action :skip_timeout, only: :seconds_until_timeout
+    # Skip these methods because they can call Devise's store_location_for, which can save timeouts path to session.
+    skip_before_action :store_current_location, :ensure_authenticated!
```

Comment:
💬 Mmm, should we add `:ensure_authenticated!` to [all lines calling `skip_before_action :store_current_location`](https://github.com/decidim/decidim/search?q=skip_before_action+%3Astore_current_location)? 
We could even create a ShouldNotStorePath concern to be more DRY. What do you think? (it's not needed to merge this, of course)

Reply:
> speech_balloon Mmm, should we add `:ensure_authenticated!` to [all lines calling `skip_before_action :store_current_location`](https://github.com/decidim/decidim/search?q=skip_before_action+%3Astore_current_location)?
> We could even create a ShouldNotStorePath concern to be more DRY. What do you think? (it's not needed to merge this, of course)

Hmmm.. Could we just remove store_location_for , from ensure_authenticated! method? It seems to me that we are saving the location twice if "Force users to authenticate before access organization" is set.

Reply:
I think you are right. It seems that that call was added in #6300 because the redirect was done before storing the current location, but I'd say that after #8036 it should work properly without the `store_location_for` call. Can you remove it and the `skip_before_action` for `ensure_authenticated!` and check if it works? Thanks!